python/dgl/distributed/optim/pytorch/sparse_optim.py

```diff
@@ -35,16 +53,208 @@ def __init__(self, params, lr):
             self._rank = 0
             self._world_size = 1
 
+    def local_state_dict(self):
+        """Return the state pertaining to current rank of the optimizer.
+
+        Returns
+        -------
+        dict
+            Local state dict
+            Example Dict of Adagrad Optimizer:
+            .. code-block:: json
+
+            {
+                "params": {
+                    "_lr": 0.01,
+                    "_eps": "1e-8",
+                    "world_size": 2
```

Comment:
Why some params have `_` prefix, some do not?

Reply:
_lr and _eps is properties of the optimizer, which need to be acquired from ```state_dict```.
world_size is not optimizer world_size, it is used in ```load``` to deal with case where Device number scale-in after reloading. As you can see, it is removed in Line 230 ``` world_size = local_state_dict[PARAMS].pop(WORLD_SIZE)``` before dumping content to current optimizer.